Userland/Libraries/LibAudio/Buffer.cpp

```diff
@@ -167,23 +167,24 @@ RefPtr<Buffer> Buffer::from_pcm_stream(InputMemoryStream& stream, ResampleHelper
 }
 
 template<typename SampleType>
-ResampleHelper<SampleType>::ResampleHelper(double source, double target)
-    : m_ratio(source / target)
+ResampleHelper<SampleType>::ResampleHelper(u32 source, u32 target)
+    : m_source(source)
+    , m_target(target)
 {
 }
-template ResampleHelper<i32>::ResampleHelper(double, double);
-template ResampleHelper<double>::ResampleHelper(double, double);
+template ResampleHelper<i32>::ResampleHelper(u32, u32);
+template ResampleHelper<double>::ResampleHelper(u32, u32);
 
 template<typename SampleType>
 Vector<SampleType> ResampleHelper<SampleType>::resample(Vector<SampleType> to_resample)
 {
     Vector<SampleType> resampled;
-    resampled.ensure_capacity(to_resample.size() * m_ratio);
+    resampled.ensure_capacity(to_resample.size() * (m_source / m_target));
```

Comment:
we have a ceil_div in StdLibExtras if that helps?
Maybe you can retain the unchecked append this way